src/uct/sm/scopy/knem/knem_md.c

```diff
@@ -204,6 +184,41 @@ static ucs_status_t uct_knem_mem_dereg(uct_md_h md,
     return status;
 }
 
+int uct_knem_md_check_mem_reg(uct_md_h md)
+{
+    uint8_t buff;
+    uct_knem_key_t key;
+
+    return (uct_knem_mem_reg_internal(md, &buff, sizeof(buff), 0, 1, &key) ==
```

Comment:
IMO better separate it:
status = reg_internal()
if (status != UCS_OK) {
return 0;
}

dereg_internal()
return 1